flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -1106,6 +1106,31 @@ class CodeGenerator(
         requireArray(array)
         generateArrayElement(this, array)
 
+      // RAND([seed])
+      case RAND if isNumeric(resultType) && operands.size <= 1 =>
```

Comment:
We don't need this changes if the method is implemented using a `CallGenerator` in `FunctionGenerator`.